float64 front_offset
float64 side_offset
